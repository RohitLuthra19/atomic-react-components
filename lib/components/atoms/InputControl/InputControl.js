// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import { InputVanilla as Input } from '../Input/Input';
import styles from './InputControl.style';

type Props = {
  id ?: string,
  type: string,
  disabled ?: boolean,
  checked ?: boolean,
  className ?: string,
};

const InputControl = ({
  id,
  type,
  checked,
  disabled,
  className,
  ...otherProps
}: Props): Node => (
  <div className={className}>
    <Input
      id={id}
      type={type}
      defaultChecked={checked}
      disabled={disabled}
      {...otherProps}
    />
    <span type={type} disabled={disabled} />
  </div>
);

InputControl.defaultProps = {
  id: '',
  checked: false,
  disabled: false,
  className: '',
};

export default styled(InputControl)`${styles}`;
export { InputControl as InputControlVanilla };
