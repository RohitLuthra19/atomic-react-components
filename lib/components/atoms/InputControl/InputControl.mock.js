
const checkedInputControl = {
  checked: true,
};

const disabledInputControl = {
  disabled: true,
};

const checkedDisabledInputControl = {
  checked: true,
  disabled: true,
};

export {
  checkedInputControl,
  disabledInputControl,
  checkedDisabledInputControl,
};
