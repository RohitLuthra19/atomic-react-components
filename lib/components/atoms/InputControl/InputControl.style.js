import { css } from 'styled-components';

const styles = css`
  display: inline-block;
  > input {
    display: none;
  }

  > span {
    border: 2px solid ${props => (props.disabled ? props.theme.fieldBorderDisabledColor : props.theme.brandPrimaryColor)};
    width: 20px;
    height: 20px;
    position: relative;
    display: inline-block;
    vertical-align: middle;

    ${props => (props.type === 'radio' && `
      border-radius: 50%;
    `)}

    &::after {
      content: "";
      width: 12px;
      height: 12px;
      left: 50%;
      top: 50%;
      transform: translate(-50%,-50%) rotate(-45deg);
      position: absolute;

      ${props => (props.type === 'checkbox' && `
        height: 6px;
        border: 2px solid #fff;
        border-width: 0 0 2px 2px;
        top: calc(50% - 3px);
      `)}

      ${props => (props.type === 'radio' && `
        border-radius: 50%;
      `)}
    }
  }

  & > input:checked + span {
    ${props => (props.type === 'radio' ? `border-color: ${props.disabled ? props.theme.fieldBorderDisabledColor : props.theme.brandPrimaryColor};` : '')}

    ${props => (props.type === 'checkbox' && `
      border-color: ${props.disabled ? props.theme.fieldBorderDisabledColor : props.theme.brandPrimaryColor};
      background-color: ${props.disabled ? props.theme.fieldBorderDisabledColor : props.theme.brandPrimaryColor};
      border-radius: 2px;
    `)}

    &:after {
      ${props => (props.type === 'radio' && `
        background-color: ${props.disabled ? props.theme.fieldBorderDisabledColor : props.theme.brandPrimaryColor};
      `)}
    }
  }
`;

export default styles;
