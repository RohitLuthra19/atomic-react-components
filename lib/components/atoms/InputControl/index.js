// @flow

export { default } from './InputControl';
