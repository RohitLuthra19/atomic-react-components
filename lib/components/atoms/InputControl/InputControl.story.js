// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

// Import Styled Component to showcase variations
import InputControl from './index';

storiesOf('Atoms', module).addWithChapters('InputControl', {
  chapters: [
    {
      title: 'Input Control - Radio',
      sections: [
        {
          sectionFn: () => <InputControl type="radio" />,
        },
        {
          sectionFn: () => <InputControl type="radio" checked />,
        },
        {
          sectionFn: () => <InputControl type="radio" disabled />,
        },
        {
          sectionFn: () => <InputControl type="radio" checked disabled />,
        },
      ],
    },
    {
      title: 'Input Control - Checkbox',
      sections: [
        {
          sectionFn: () => <InputControl type="checkbox" />,
        },
        {
          sectionFn: () => <InputControl type="checkbox" checked />,
        },
        {
          sectionFn: () => <InputControl type="checkbox" disabled />,
        },
        {
          sectionFn: () => <InputControl type="checkbox" checked disabled />,
        },
      ],
    },
  ],
});
