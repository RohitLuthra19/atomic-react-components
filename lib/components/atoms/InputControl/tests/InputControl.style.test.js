import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import InputControl from '../InputControl';
import Theme from '../../../../styles/theme';

describe('InputControl', () => {
  let inputControl = '';
  const theme = { ...Theme };

  test('should render radio correctly', () => {
    inputControl = shallow(<InputControl theme={theme} type="radio" />);
    expect(inputControl).toMatchSnapshot();
  });

  test('should render checkbox correctly', () => {
    inputControl = shallow(<InputControl theme={theme} type="checkbox" />);
    expect(inputControl).toMatchSnapshot();
  });
});
