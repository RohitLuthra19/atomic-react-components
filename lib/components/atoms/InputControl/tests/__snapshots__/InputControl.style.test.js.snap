// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputControl should render checkbox correctly 1`] = `
.c0 {
  display: inline-block;
}

.c0 > input {
  display: none;
}

.c0 > span {
  border: 2px solid #006193;
  width: 20px;
  height: 20px;
  position: relative;
  display: inline-block;
  vertical-align: middle;
}

.c0 > span::after {
  content: "";
  width: 12px;
  height: 12px;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%) rotate(-45deg);
  -ms-transform: translate(-50%,-50%) rotate(-45deg);
  transform: translate(-50%,-50%) rotate(-45deg);
  position: absolute;
  height: 6px;
  border: 2px solid #fff;
  border-width: 0 0 2px 2px;
  top: calc(50% - 3px);
}

.c0 > input:checked + span {
  border-color: #006193;
  background-color: #006193;
  border-radius: 2px;
}

<InputControl
  checked={false}
  className="c0"
  disabled={false}
  id=""
  theme={
    Object {
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "borderColorBoldest": "#c00",
      "borderColorSecondary": "#d9d9d9",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "fieldBackgroundColor": "#ffffff",
      "fieldBackgroundDisabledColor": "#f3f3f3",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fieldTextColor": "#222",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "fontSizeHighlight": "18px",
      "letterSpacing": "0px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "linkColorSecondary": "#016fad",
      "maxWidthPageContent": "80rem",
      "textColorBrandPrimary": "#006193",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSecondaryLight": "#666",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zLoader": 1000,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zSticky": 700,
      },
    }
  }
  type="checkbox"
/>
`;

exports[`InputControl should render radio correctly 1`] = `
.c0 {
  display: inline-block;
}

.c0 > input {
  display: none;
}

.c0 > span {
  border: 2px solid #006193;
  width: 20px;
  height: 20px;
  position: relative;
  display: inline-block;
  vertical-align: middle;
  border-radius: 50%;
}

.c0 > span::after {
  content: "";
  width: 12px;
  height: 12px;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%) rotate(-45deg);
  -ms-transform: translate(-50%,-50%) rotate(-45deg);
  transform: translate(-50%,-50%) rotate(-45deg);
  position: absolute;
  border-radius: 50%;
}

.c0 > input:checked + span {
  border-color: #006193;
}

.c0 > input:checked + span:after {
  background-color: #006193;
}

<InputControl
  checked={false}
  className="c0"
  disabled={false}
  id=""
  theme={
    Object {
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "borderColorBoldest": "#c00",
      "borderColorSecondary": "#d9d9d9",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "fieldBackgroundColor": "#ffffff",
      "fieldBackgroundDisabledColor": "#f3f3f3",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fieldTextColor": "#222",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "fontSizeHighlight": "18px",
      "letterSpacing": "0px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "linkColorSecondary": "#016fad",
      "maxWidthPageContent": "80rem",
      "textColorBrandPrimary": "#006193",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSecondaryLight": "#666",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zLoader": 1000,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zSticky": 700,
      },
    }
  }
  type="radio"
/>
`;
